File: 프로그래머스/3/43162. 네트워크/네트워크.py


def solution(n, computers):
    visited = [0]*(n+1)
    
    def recur(node):
        visited[node] =1
        for nxt in range(n):
            if visited[nxt] ==0 and computers[node][nxt] == 1:
                recur(nxt)
    
    count = 0
    for i in range(n):
        if visited[i] == 0:
            recur(i)
            count +=1
            
    return count

    

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

vector<bool> visited;
vector<vector<int>> graph;

void recur(int node, int n) {
    visited[node] = true;
    for (int nxt = 0; nxt<n; nxt++){
        if (!visited[nxt] && graph[node][nxt] == 1){
            recur(nxt, n);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    visited.assign(n, false);
    graph = computers;
    
    int answer = 0;
    for (int i=0; i<n; i++){
        if (!visited[i]) {
            recur(i,n);
            answer++;
        }
    }
    return answer;
}